### PR TITLE
[11.0][MIG] account_payment_sale

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: account_payment_sale/README.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+=============
+
+There is nothing to configure.
+
+Usage
+=====
+
+You are able to add a payment mode directly on a partner.
+This payment mode is automatically associated to the sale order, then on related invoice. 
+This default value could be change in a draft sale order or draft invoice.
```
Comment (simahawk) [inline account_payment_sale/README.rst]: `s/could be change/can be changed`

File: account_payment_sale/README.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/173/9.0
```
Comment (simahawk) [inline account_payment_sale/README.rst]: `/11.0`

File: account_payment_sale/README.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+Known issues / Roadmap
```
Comment (simahawk) [inline account_payment_sale/README.rst]: remove section if empty

File: account_payment_sale/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    'summary': "Adds payment mode on sale orders",
+    'author': "Akretion, "
+              "Tecnativa, "
+              "Odoo Community Association (OCA)",
+    'website': 'https://github.com/OCA/bank-payment',
+    'depends': [
+        'sale',
+        'account_payment_partner',
+    ],
+    'conflicts': ['sale_payment'],
```
Comment (simahawk) [inline account_payment_sale/__manifest__.py]: @pedrobaeza is "conflicts" considered somewhere?

File: account_payment_sale/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    'data': [
+        'views/sale_order_view.xml',
+        'views/sale_report_templates.xml',
+    ],
+    'installable': True,
```
Comment (simahawk) [inline account_payment_sale/__manifest__.py]: you can remove this flags

File: account_payment_sale/views/sale_report_templates.xml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <template id="report_sale_payment_mode" inherit_id="sale.report_saleorder_document">
+        <xpath expr="//p[@t-if='doc.payment_term_id.note']" position="after">
+            <p t-if="doc.payment_mode_id.note">
```
Comment (simahawk) [inline account_payment_sale/views/sale_report_templates.xml]: add `id="payment_mode_note"` to ease customization

File: account_payment_sale/models/sale_order.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+            self.payment_mode_id = self.partner_id.customer_payment_mode_id
+        else:
+            self.payment_mode_id = False
+        return res
+
+    @api.multi
+    def _prepare_invoice(self):
+        """Copy bank partner from sale order to invoice"""
+        vals = super(SaleOrder, self)._prepare_invoice()
+        if self.payment_mode_id:
```
Comment (simahawk) [inline account_payment_sale/models/sale_order.py]: I'd split this to `_get_payment_mode_vals` method and reuse it here and in the wizard

File: account_payment_sale/wizard/sale_make_invoice_advance.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+class SaleAdvancePaymentInv(models.TransientModel):
+    _inherit = 'sale.advance.payment.inv'
+
+    @api.multi
+    def _create_invoice(self, order, so_line, amount):
+        """Copy payment mode from sale order to invoice"""
+        inv = super(SaleAdvancePaymentInv, self)._create_invoice(
+            order, so_line, amount)
+        vals = {}
+        if order.payment_mode_id:
```
Comment (simahawk) [inline account_payment_sale/wizard/sale_make_invoice_advance.py]: see prev comment on `_get_payment_mode_vals`

**Review by simahawk — CHANGES_REQUESTED** [review]

File: account_payment_sale/__manifest__.py (hunk 4 above)
Comment (pedrobaeza) [reply account_payment_sale/__manifest__.py]: No, it's only informative, and this doesn't make sense right now. It was for v7...

File: account_payment_sale/README.rst [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+=======
+
+Contributors
+------------
+
+* Pedro M. Baeza
+* Alexis de Lattre <alexis.delattre@akretion.com>
+* Alexandre Fayolle
+* Danimar Ribeiro
+* Raphaël Valyi
```
Comment (mpanarin) [inline account_payment_sale/README.rst:61]: add contributor )

**Review by mpanarin — APPROVED** [review]
LGTM

**Review by simahawk — APPROVED** [review]

**Review by grindtildeath — APPROVED** [review]

**Pull request merged**